internal/input/accounts.go

```diff
@@ -0,0 +1,69 @@
+package input
+
+import (
+	"context"
+
+	"github.com/stellar/stellar-etl/internal/utils"
+
+	"github.com/stellar/go/exp/ingest/adapters"
+	ingestio "github.com/stellar/go/exp/ingest/io"
+	"github.com/stellar/go/historyarchive"
+	"github.com/stellar/go/xdr"
+)
+
+var archiveStellarURL = "http://history.stellar.org/prd/core-live/core_live_001"
+
+// GetAccountsFromGenesis returns a slice of ledger close metas for the ledgers starting from the genesis ledger and ending at end (inclusive)
+func GetAccountsFromGenesis(end uint32) ([]xdr.LedgerEntry, error) {
+	archive, err := historyarchive.Connect(
+		archiveStellarURL,
+		historyarchive.ConnectOptions{Context: context.Background()},
+	)
+	if err != nil {
+		return []xdr.LedgerEntry{}, err
+	}
+
+	historyAdapter := adapters.MakeHistoryArchiveAdapter(archive)
+	latestNum, err := historyAdapter.GetLatestLedgerSequence()
+	if err != nil {
+		return []xdr.LedgerEntry{}, err
+	}
+
+	err = validateLedgerRange(1, end, latestNum)
+	if err != nil {
+		return []xdr.LedgerEntry{}, err
+	}
+
+	checkpointSeq, err := utils.GetCheckpointNum(end, latestNum)
+	if err != nil {
+		return []xdr.LedgerEntry{}, err
+	}
+
+	return readBucketList(archive, checkpointSeq, xdr.LedgerEntryTypeAccount)
+}
+
+func readBucketList(archive *historyarchive.Archive, checkpointSeq uint32, entryType xdr.LedgerEntryType) ([]xdr.LedgerEntry, error) {
```

Comment:
I'd consider pulling out this function into a separate file, as other CLI's might also need it. That can be done in a subsequent PR for sure.